lib/httpClientToRequestPolicyAdapter.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import { HttpClient } from "./httpClient";
+import { HttpRequest } from "./httpRequest";
+import { HttpResponse } from "./httpResponse";
+import { RequestPolicy } from "./requestPolicy";
+
+/**
+ * An adapter type that adapts a HttpClient to look like a RequestPolicy.
+ */
+export class HttpClientToRequestPolicyAdapter implements RequestPolicy {
```

Comment:
Do adapters still need to be a thing when we have structural typing? that is, it seems like HttpClient should be assignable to RequestPolicy in this language

Reply:
I guess RequestPolicy has a named method on it, so not quite. This could maybe be accomplished with something like:

```js
{
    send: httpClient
}
```
but this is bikeshedding a bit

Reply:
I changed HttpClient to be a function type instead of an interface. It makes testing easier and I'm hoping it will make it simpler to create new HttpClients (for the handful of people who want to do that). However, RequestPolicy is still an interface so there needs to be an adapter. I could consider changing RequestPolicy to also be a function type, in which case an adapter wouldn't be necessary...

Reply:
Turns out if I change RequestPolicy to have an invokable method on it, then BaseRequestPolicy can't be abstract. It would have to implement the invokable method. That seems to be more trouble than it's worth, so I'm just going to change HttpClient back to being an interface with a send() method on it.

Reply:
Now HttpClient implements RequestPolicy, so the adapter goes away. Problem solved.

Reply:
Given that, still worth considering using structural typing at the usage site of this adapter https://github.com/Azure/ms-rest-js/pull/33#discussion_r179842294 instead of declaring a named adapter class.

That is, `{ send: httpClient }` instead of `new HttpClientToRequestPolicyAdapter(httpClient)`